packages/experience-builder-sdk/src/hooks/useFetchExperience.test.ts

```diff
@@ -26,51 +27,44 @@ describe('useFetchExperience', () => {
     } as unknown as ContentfulClientApi<undefined>;
   });
 
-  it('should be defined', () => {
-    const res = renderHook((props) => useFetchExperience(props), {
-      initialProps: { client: clientMock, mode: 'preview' as ExternalSDKMode },
+  it('should be defined', async () => {
+    const { result } = await act(async () => {
+      return renderHook(useFetchExperience, {
+        initialProps: {
+          client: clientMock,
+          mode: 'preview' as ExternalSDKMode,
+          slug,
+          experienceTypeId,
+          localeCode,
+        },
+      });
```

Comment:
Wrapping `render` statements inside `act` is considered a bad practice afaik. I can recommend reading [this](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily) and the linked blog post.
TLDR; In the first render cycle, it fetches the experience and thus has an async behaviour. React notices this which could harm the integrity of our unit tests and thus probably printed a warning in your case. To address this, the test setup should make sure that all async this settled before running any further action or assertion. For that you can usually use something like `waitFor` that will wait for something to change in the returned object that signals you that it's not fetching anymore.

Reply:
thanks! TIL. Updating to wrap the  expects in a waitFor